ui/test/nuxeo-aggregation-widget.test.js

```diff
@@ -0,0 +1,364 @@
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible></nuxeo-checkbox-aggregation>
+        `,
+      );
+      await flush();
+
+      const nuxeoCollapsible = nuxeoCheckboxAggregation.shadowRoot.querySelector('nuxeo-collapsible');
+      const nuxeoCollapsibleButton = nuxeoCollapsible.shadowRoot.querySelector('button');
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(8);
+      checkboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.false;
+      });
+
+      tap(nuxeoCollapsibleButton);
+      expect(checkboxes.length).to.equal(8);
+      for (let i = 0; checkboxes.length <= 7; i++) {
+        expect(isElementVisible(checkboxes[i])).to.be.true;
+      }
+      expect(isElementVisible(checkboxes[8])).to.be.false;
+    });
+
+    test('Should display 4 checkboxes in a collapsible element when "visibleItems" is set to 4', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible visible-items="4"></nuxeo-checkbox-aggregation>
+        `,
+      );
+      await flush();
+
+      const nuxeoCollapsible = nuxeoCheckboxAggregation.shadowRoot.querySelector('nuxeo-collapsible');
+      const nuxeoCollapsibleButton = nuxeoCollapsible.shadowRoot.querySelector('button');
+      const checkboxes = nuxeoCheckboxAggregation.shadowRoot.querySelectorAll('paper-checkbox');
+      expect(checkboxes.length).to.equal(4);
+      checkboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.false;
+      });
+
+      tap(nuxeoCollapsibleButton);
+      checkboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.true;
+      });
+    });
+
+    test('Should show 9 checkboxes when "visibleItems" is set to 4 and "show more" button is toggled', async () => {
+      const nuxeoCheckboxAggregation = await fixture(
+        html`
+          <nuxeo-checkbox-aggregation .data=${data} collapsible visible-items="4"></nuxeo-checkbox-aggregation>
+        `,
+      );
+      await flush();
+      const nuxeoCollapsible = nuxeoCheckboxAggregation.shadowRoot.querySelector('nuxeo-collapsible');
+      const nuxeoCollapsibleButton = nuxeoCollapsible.shadowRoot.querySelector('button');
+      const slotContent = nuxeoCollapsible.shadowRoot.querySelectorAll('slot')[1];
+      let paperCheckboxes = selectElementInSlot(slotContent, 'paper-checkbox');
+      expect(paperCheckboxes.length).to.equal(4);
+      paperCheckboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.false;
+      });
+
+      tap(nuxeoCollapsibleButton);
+      expect(paperCheckboxes.length).to.equal(4);
+      paperCheckboxes.forEach((element) => {
+        expect(isElementVisible(element)).to.be.true;
+      });
```

Comment:
We are checking this pattern a lot of times. Wondering if we may want to drop the testing part of it, or trying to make it as part of a setup or maybe extracting that logic to class helper.

Reply:
Do you feel comfortable with shipping as is and creating a ticket for that? Or do you think this should be tackled now?

Reply:
I would say that it should be tackled now, since it would be shipped in this file context 🤷‍♂️

Reply:
will do it